### PR TITLE
feat(backend): update listing filters

### PR DESCRIPTION
**Pull request by pbn4**
## Issue

- Closes #issue
- Addresses #1015 
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds `isVerified` filter param and fixes how community programs are queried.

## How Can This Be Tested/Reviewed?

Requires frontend updates

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend/core/src/shared/query-filter/custom_filters.ts [hunk 1]
```diff
@@ -5,46 +5,6 @@ import {
...
-    qb.andWhere(
-      `(${reservedCommunityTypeColumnName} IS NULL OR ${reservedCommunityTypeColumnName} <> LOWER(:${whereParameterName}))`,
-      {
-        [whereParameterName]: independentLivingCommunityType,
-      }
-    )
-  }
-}
-
 export function addAvailabilityQuery(
```
Comment (emilyjablonski) [inline backend/core/src/shared/query-filter/custom_filters.ts:8]: I could be missing this but I think we need another query here for bedroom type based on the updates made for unit groups - something like, "does this bedroom type existing in any of the unit groups on this listing"

**Comment by YazeedLoonat** [comment]
Hey @emilyjablonski could you take another look at this I think this is okay for the backend and is ready for the front end

File: backend/core/src/shared/query-filter/custom_filters.ts (hunk 1 above)
Comment (pbn4) [reply backend/core/src/shared/query-filter/custom_filters.ts:8]: I confirmed how it works right now when you provide following filters:

```
filter[0][$comparison]: =
filter[0][bedrooms]: 3
```

it returns a listing but attached are only those unit groups with bedrooms number equal to 3, which is not what is expected from this filter unfortunately. I need to do some more work.

**Comment by emilyjablonski** [comment]
@YazeedLoonat Do you mind reviewing Michal's latest commit?

**Review by seanmalbert — CHANGES_REQUESTED** [review]
@pbn4 ,
`addBedroomsQuery` should be able to accept an array of numbers, so if I provide a filter of:
```
filter[0][$comparison]: IN
filter[0][bedrooms]: 1,3
```
it should work.

You can probably think of a better alternative, but I got that working locally by doing the following:
1. update filterValue to be a type of `number[]`
2. update the select to `num_bedrooms IN (:...${whereParameterName})`
3. update the call to be
```
addBedroomsQuery(
  qb,
  typeof filterValue === "string"
    ? filterValue.split(",").map((val) => Number(val))
    : [filterValue]
)
```

**Comment by pbn4** [comment]
> You can probably think of a better alternative, but I got that working locally by doing the following:

It looks fine to me.

**Pull request merged**